src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/HostedMcpServerToolResultContent.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Represents the result of a hosted MCP server tool call.
+/// </summary>
+public class HostedMcpServerToolResultContent : AIContent
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="HostedMcpServerToolResultContent"/> class.
+    /// </summary>
+    /// <param name="callId">The tool call ID.</param>
+    public HostedMcpServerToolResultContent(string callId)
+    {
+        CallId = Throw.IfNullOrWhitespace(callId);
+    }
+
+    /// <summary>
+    /// Gets the tool call ID.
+    /// </summary>
+    public string CallId { get; }
+
+    /// <summary>
+    /// Gets or sets the output of the tool call.
+    /// </summary>
+    public IList<AIContent>? Output { get; set; }
+
+    /// <summary>
+    /// Gets a value indicating whether the tool call was successful.
+    /// </summary>
+    public virtual bool Success => Output is not null && Output.Count is not 0 && !Output.OfType<ErrorContent>().Any();
```

Comment:
Do we need this property?

And why is it virtual?

Reply:
> I added it to provide a reliable way of determining if the result was successful. Are you suggesting that we should not do that?

I don't think we need it, and we can always add it later if we do. Let's keep things as minimal as possible until we learn more.